pkg/util/loop/loop.go

```diff
@@ -160,37 +179,36 @@ func (loop *Device) AttachFromFile(image *os.File, mode int, number *int) error
 				return nil
 			}
 			syscall.Close(loopFd)
+			continue
 		} else {
 			_, _, esys := syscall.Syscall(syscall.SYS_IOCTL, uintptr(loopFd), CmdSetFd, image.Fd())
 			if esys != 0 {
 				syscall.Close(loopFd)
 				continue
 			}
-			break
 		}
-	}
 
-	if _, _, err := syscall.Syscall(syscall.SYS_FCNTL, uintptr(loopFd), syscall.F_SETFD, syscall.FD_CLOEXEC); err != 0 {
-		return fmt.Errorf("failed to set close-on-exec on loop device %s: %s", path, err.Error())
-	}
+		if _, _, err := syscall.Syscall(syscall.SYS_FCNTL, uintptr(loopFd), syscall.F_SETFD, syscall.FD_CLOEXEC); err != 0 {
+			return fmt.Errorf("failed to set close-on-exec on loop device %s: %s", path, err.Error())
+		}
 
-	maxRetries := 5
-	for i := 0; i < maxRetries; i++ {
 		if _, _, err := syscall.Syscall(syscall.SYS_IOCTL, uintptr(loopFd), CmdSetStatus64, uintptr(unsafe.Pointer(loop.Info))); err != 0 {
-			if err == syscall.EAGAIN && i < maxRetries-1 {
-				// with changes introduces in https://github.com/torvalds/linux/commit/5db470e229e22b7eda6e23b5566e532c96fb5bc3
-				// loop_set_status() can temporarily fail with EAGAIN -> sleep and try again
-				// (cf. https://github.com/karelzak/util-linux/blob/dab1303287b7ebe30b57ccc78591070dad0a85ea/lib/loopdev.c#L1355)
-				time.Sleep(250 * time.Millisecond)
+			if err == syscall.EAGAIN {
+				eagainFound = true
 				continue
 			}
 			// clear associated file descriptor to release the loop device,
 			// best-effort here without error checking because we need the
 			// error from previous ioctl call
 			syscall.Syscall(syscall.SYS_IOCTL, uintptr(loopFd), CmdClrFd, 0)
 			return fmt.Errorf("failed to set loop flags on loop device: %s", syscall.Errno(err))
+		} else {
+			break
 		}
-		break
+	}
+
+	if eagainFound {
+		return syscall.EAGAIN
 	}
```

Comment:
The simple boolean and Errno return value hides the complexity that there was a specific device or devices that we were asked to retry (though perhaps it would be too late to retry them after we have released them, if we fix the above issue).